features/dd-sdk-android-session-replay/src/test/kotlin/com/datadog/android/sessionreplay/internal/domain/RequestBodyFactoryTest.kt

```diff
@@ -46,76 +47,66 @@ internal class RequestBodyFactoryTest {
     @Mock
     lateinit var mockCompressor: BytesCompressor
 
-    lateinit var fakeCompressedData: ByteArray
+    lateinit var fakeGroupedSegments: List<Pair<MobileSegment, JsonObject>>
 
-    @Forgery
-    lateinit var fakeSegment: MobileSegment
-
-    @Forgery
-    lateinit var fakeSegmentAsJson: JsonObject
-
-    lateinit var fakeSerializedSegmentWithNewLine: String
+    lateinit var fakeCompresseData: List<ByteArray>
 
     @BeforeEach
     fun `set up`(forge: Forge) {
-        fakeSerializedSegmentWithNewLine = fakeSegmentAsJson.toString() + "\n"
-        fakeCompressedData = forge.aString().toByteArray()
-        whenever(mockCompressor.compressBytes(fakeSerializedSegmentWithNewLine.toByteArray()))
-            .thenReturn(fakeCompressedData)
+        fakeGroupedSegments = forge.aList(size = forge.anInt(min = 1, max = 10)) {
```

Comment:
minor: alternatively you can also use `@MapForgery` to create a map of `<MobileSegment, JsonObject>`